faith_presentation/src/commonMain/kotlin/net/thechance/mena/faith/presentation/feature/quran/bookmark/BookmarkScreen.kt

```diff
@@ -0,0 +1,182 @@
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.statusBarsPadding
+import androidx.compose.foundation.layout.systemBarsPadding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import app.cash.paging.compose.LazyPagingItems
+import app.cash.paging.compose.collectAsLazyPagingItems
+import mena.faith_presentation.generated.resources.Res
+import mena.faith_presentation.generated.resources.bookmarks
+import mena.faith_presentation.generated.resources.empty_state_bookmark_description
+import mena.faith_presentation.generated.resources.empty_state_bookmark_image
+import mena.faith_presentation.generated.resources.empty_state_bookmark_title
+import mena.faith_presentation.generated.resources.ic_not_saved_book_mark
+import net.thechance.mena.designsystem.presentation.component.appBar.AppBar
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import net.thechance.mena.faith.presentation.base.ObserveAsEffect
+import net.thechance.mena.faith.presentation.base.SnackBarState
+import net.thechance.mena.faith.presentation.component.BackIcon
+import net.thechance.mena.faith.presentation.component.FaithScaffold
+import net.thechance.mena.faith.presentation.component.FaithSnackBar
+import net.thechance.mena.faith.presentation.component.SwappableCard
+import net.thechance.mena.faith.presentation.designSystem.theme.QuranTheme
+import net.thechance.mena.faith.presentation.extensions.paging.isEmpty
+import net.thechance.mena.faith.presentation.extensions.paging.isNotEmpty
+import net.thechance.mena.faith.presentation.feature.quran.bookmark.component.AyaBookmarkCard
+import net.thechance.mena.faith.presentation.feature.quran.bookmark.component.EmptyBookmarkState
+import net.thechance.mena.faith.presentation.navigation.LocalNavController
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.koin.compose.viewmodel.koinViewModel
+
+@Composable
+fun BookmarkScreen(
+    viewModel: BookmarkViewModel = koinViewModel()
+) {
+
+    val state by viewModel.uiState.collectAsStateWithLifecycle()
+    val snackBarState by viewModel.snackBarState.collectAsStateWithLifecycle()
+    val navController = LocalNavController.current
+
+    ObserveAsEffect(viewModel.uiEffect) { effect ->
+        when (effect) {
+            is BookmarkEffect.NavigateBack -> {
+                navController.navigateUp()
+            }
```

Comment:
What's hard about this?

```suggestion
            is BookmarkEffect.NavigateBack -> navigateUp()
```

Reply:
My real name is Fares Lambda 😂 
I will check all lambdas like this and fix it ❤️